pkg/controllers/management/auth/user.go

```diff
@@ -0,0 +1,235 @@
+package auth
+
+import (
+	"fmt"
+
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+type userLifecycle struct {
+	projectRoles        v3.ProjectRoleTemplateBindingInterface
```

Comment:
Can you name all these properties to match the actual type that they are for? ClusterRole and GlobalRole are actual resources, so not having  binding in the name will lead to confusion